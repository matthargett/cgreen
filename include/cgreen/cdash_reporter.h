#ifndef CDASH_REPORTER_HEADER
#define CDASH_REPORTER_HEADER

#include <cgreen/reporter.h>
#include <stdio.h>

#ifdef __cplusplus
namespace cgreen {
    extern "C" {
#endif

typedef struct CDashInfo_ CDashInfo;
struct CDashInfo_ {
	char *name;
	char *build;
	char *type;
	char *hostname;
	char *os_name;
	char *os_platform;
	char *os_release;
	char *os_version;
};

typedef int CDashPrinter(FILE *, const char *format, ...);

extern void set_cdash_printer(TestReporter *reporter, CDashPrinter *printer);
extern TestReporter *create_cdash_reporter(CDashInfo *cdash);

#ifdef __cplusplus
    }
}
#endif

#endif
