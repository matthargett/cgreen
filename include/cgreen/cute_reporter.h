#ifndef CUTE_REPORTER_HEADER
#define CUTE_REPORTER_HEADER

#include <cgreen/reporter.h>

#ifdef __cplusplus
namespace cgreen {
    extern "C" {
#endif

typedef int CutePrinter(const char *format, ...);


extern TestReporter *create_cute_reporter(void);
extern void set_cute_printer(TestReporter *reporter, CutePrinter *printer);

#ifdef __cplusplus
    }
}
#endif

#endif
