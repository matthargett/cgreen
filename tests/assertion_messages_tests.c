#include <cgreen/cgreen.h>
#include <cgreen/constraint.h>

#include <stdio.h>
#include <stdlib.h>
#include <unistd.h>

#ifdef __cplusplus
using namespace cgreen;
#endif

Describe(AssertionMessage);
BeforeEach(AssertionMessage) {}
AfterEach(AssertionMessage) {}


char something[42];
Ensure(AssertionMessage, for_comparing_content_with_negative_length) {
    assert_that(47, is_equal_to_contents_of(something, -4));
}
