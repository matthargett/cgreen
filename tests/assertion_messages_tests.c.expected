Running "assertion_messages_tests" (1 tests)...
assertion_messages_tests.c: Failure: AssertionMessage -> for_comparing_content_with_negative_length 
			Wanted to compare contents with [something], but [-4] was given for the comparison size.

Completed "AssertionMessage": 0 passes, 1 failure, 0 exceptions in 0ms.
Completed "assertion_messages_tests": 0 passes, 1 failure, 0 exceptions in 0ms.
